### ISSUE TITLE
Reporting: SFO Final Report Review Page

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-reporting#149]
#### Description
Before Industrial Reporters get to the "submit report" page, we need to show them a final review page that displays all the fields they have input into the system 
This allows them to print and share the full report before submitting. 

#### Acceptance Criteria
Given I am an Industrial Reporter who has completed my annual report
Before I am sent to the submit page, I want to see a full list of all fields for my annual report
So that I can review them before submission. 

As an SFO, Given I am on the final review page, I see the following in read-only mode
[SFO wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=25479-6789&t=Z9nk3eWKE8pnB7dZ-4)

In the Final Review section of the task list, hide the “sign off and submit” section because the user is already in that section of the report. 

The user can click the following buttons: Back, Continue
The "Back button" will take the user to the Compliance Summary page
The "Continue button" will take the user to the Verification page

#### Development Checklist:

- [ ] Create review-submit page similar to `/bciers/apps/reporting/src/app/bceidbusiness/industry_user/reports/[version_id]/review-operator-data`
- [ ] Create RJSF Schema with all steps and foldable groups
- [ ] Create UISchema
- [ ] (Maybe create accordion widget or edit css depending on review)
- [ ] Link redirect on previous compliance summary review page 
- [ ] Perhaps create endpoint/service to grab all report data in one blob for RJSF if needed
- [ ] Add tests for backend endpoint
- [ ] Add tests for frontend page
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by rdromey** [comment]
@gdalcengio to write dev checklist

**Comment by patriciarussellCAS** [comment]
Consider reviewing what is done in CIIP for similar feature

**Comment by patriciarussellCAS** [comment]
Final design still pending approval

**Comment by patriciarussellCAS** [comment]
To confirm with Devs: do we need additional tickets for the other reports? a final review page for Simple Reports, a final review page for Reporting Operations and EIOs?

**Comment by patriciarussellCAS** [comment]
@zoeyli-46 see my comment on figma re: SFO report view.

**Comment by zoeyli-46** [comment]
Adjusted wireframe and the AC on this ticket

**Comment by rdromey** [comment]
@pbastia to split into two tickets

**Comment by pbastia** [comment]
@rdromey #420 is the first (albeit very very small) part for this ticket

**Comment by pbastia** [comment]
@ayeshmcg idea: We could probably do a single query with all dependent tables: 
```python
report_version = ReportVersion.objects.select_related("facility_report", "report_operation", "facility_report__activity_report", ...).get(report_version=<xyz>)
```

and let the existing ninja schema objects put together a big json object that has everything every single form page needs

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/ER1noH2xX2VDrXf7QpViyI8B5cfXvZhQAbq_uSvV9ot6hw?e=htGHQf).

I can verify that:
- [x] Before I am sent to the Submit page, I can see the full list of fields for my annual report, except for ```Fuel combustion by mobile equipment``` (blocked by #518)
- [x] In the Final Review page, I see the Sign-off  and Submit section only in the task list
- [x] The user is able to click the ```Back``` and ```Continue``` buttons
- [x] The ```Back``` button redirects to the ```Compliance Summary``` page
- [x] The ```Continue``` button redirects to the ```Verification``` page (if Verification is needed)
- [ ] The ```Continue``` button redirects to the ```Attachments``` page (if Verification is not needed) - **current behavior is that it redirects to the Verification page, see page 23 of the document**

Additional notes:
- In the Final Review page, all data in the Allocation of Emissions were correct but it can be difficult to read. Suggest to fix the alignment but that can be in another ticket if needed.
- In the Final Review page, under Non-Attributable Emissions, the radio buttons in the first question ```Did your non-attributable emissions exceed 100 tCO2e?``` are non-clickable, but seems like they can be because they don't look disabled. Suggest to also make it look disabled if possible.

Moving to Pending PO for review of test and notes, cc: @patriciarussellCAS / @vesselak

**Comment by patriciarussellCAS** [comment]
@zoeyli-46 did you want to do a design review of this? Would love your thoughts.

**Comment by zoeyli-46** [comment]
**Design Review**
**Screenshot of issue**
<img width="591" alt="Image" src="https://github.com/user-attachments/assets/7186d30e-2382-4796-83cc-08dd1d6126cb" />

- Could we hide the bottom divider for the last item before an accordion, and remove the top divider for the first item under an accordion? So it can look like this instead: 
**Corrected screen**
<img width="800" alt="Image" src="https://github.com/user-attachments/assets/95f9da2d-44fc-41f0-a7fc-6ad1edb914ea" />

This issue doesn't appear on every accordion, for example on the GSC accordion, the divider only shows up on the item above the accordion but not the one below it
**Screenshot of issue**
<img width="800" alt="Image" src="https://github.com/user-attachments/assets/d3450067-2a34-425c-bb60-c7f3decf3735" />

Another example of the issue:
**Screenshot of issue**
- there's an extra divider above emissions summary
<img width="800" alt="Image" src="https://github.com/user-attachments/assets/0e2ef33c-b0a6-4a5f-8d34-ee33ac09e63e" />

**Comment by zoeyli-46** [comment]
**Design review:
Screenshot of issue**
- Could we remove the dividers in the emissions section
- Could we make the emissions section width not exceed the width of the fuel/unit card? it currently extends past the card and juts out like this
- methodology text alignment is different from the rest of the fields, perhaps the spacing between field label and input is different? or the width of field label is shorter than the others?
<img width="983" alt="Image" src="https://github.com/user-attachments/assets/3206d8c6-b8fe-4cf0-8c1f-caba0d8171b4" />

**Comment by zoeyli-46** [comment]
**Design review:
Screenshot of issue:**
- On Allocation of Emissions page, "equivalent" fields are not aligned with the rest of the form's field inputs. perhaps the spacing between field label and input is different? or the width of field label is shorter than the others?
<img width="490" alt="Image" src="https://github.com/user-attachments/assets/1ca93d82-6f52-401e-83eb-3940f35c23e3" />

**Screenshot of issue**
- On Compliance Summary page, the field inputs are not aligned
<img width="490" alt="Image" src="https://github.com/user-attachments/assets/bdc0392d-16e4-4635-adcb-109c35da62a2" />
